doc/advanced_configuration.rst

```diff
@@ -51,10 +51,10 @@ you would do:
 .. code-block:: python
 
     sphinx_gallery_conf = {
-        'filename_pattern' : 'plot_awesome_example.py'
+        'filename_pattern' : 'plot_awesome_example\.py'
     }
 
-Similarly, to build only examples in a specific directory, you can do:
+Here, we escape ``'.'`` because it is a special character in `regular expressions`_. Similarly, to build only examples in a specific directory, you can do:
```

Comment:
Maybe I would rather leave the example without the backslash and add a note saying that if you want to be pedantic you can escape it but it will work for 99.99 % of the use case if not more.